store/sqlstore/compliance_store.go

```diff
@@ -151,17 +149,38 @@ func (s SqlComplianceStore) ComplianceExport(job *model.Compliance) ([]*model.Co
 			Channels,
 			Users,
 			Posts
-        LEFT JOIN Bots ON Bots.UserId = Posts.UserId
+		LEFT JOIN
+			Bots ON Bots.UserId = Posts.UserId
 		WHERE
 			Teams.Id = Channels.TeamId
 				AND Posts.ChannelId = Channels.Id
 				AND Posts.UserId = Users.Id
-				AND Posts.CreateAt > :StartTime
+				AND Posts.CreateAt > :LastPostCreateAt
 				AND Posts.CreateAt <= :EndTime
 				` + emailQuery + `
-				` + keywordQuery + `)
-		UNION ALL
-		(SELECT
+				` + keywordQuery + `
+		ORDER BY PostCreateAt
+		LIMIT :Limit`
+		if _, err := s.GetReplica().Select(&channelPosts, channelsQuery, props); err != nil {
+			return nil, errors.Wrap(err, "unable to export compliance")
+		}
+		if len(channelPosts) < limit {
+			cursor.ChannelsQueryCompleted = true
+		} else {
+			cursor.LastChannelsQueryPostCreateAt = channelPosts[len(channelPosts)-1].PostCreateAt
+		}
+	}
+
+	var directMessagePosts []*model.CompliancePost
+	directMessagesQuery := ""
+	if !cursor.DirectMessagesQueryCompleted && len(channelPosts) < limit {
```

Comment:
The idea behind this clause is to return rows from two tables in one function call.
Let's say there are 3 channel posts, and 3 DMs. The caller is retrieving posts in batches of 2.
The first call will return 2 channel posts; the second call will return 1 channel post and 1 DM; the third call will return 2 DMs.

Reply:
I know , I just meant because above you do 
```
		if len(channelPosts) < limit {
			cursor.ChannelsQueryCompleted = true
		}
```
then you don't need to recheck the condition and just use the variable which has a name to it and makes it more clean what len(channelPosts) < limit really means. that's what I was suggesting here.

Reply:
I see. I personally think that `len(channelPosts) < limit` is clearer because it shows that we are trying to "fill up" the limit specified by the caller.